File: Jumping_on_the_Clouds_Revisited.js
'use strict';

const fs = require('fs');

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.replace(/\s*$/, '')
        .split('\n')
        .map(str => str.replace(/\s*$/, ''));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the jumpingOnClouds function below.
function jumpingOnClouds(c, k) {
    let n= c.length;
    let e=100;
    if(n==k){
      if(c[0]==1){
          e=e-3;
      }  
      else{
          e=e-1;
      }
    }
    else{
        for(let i=k;i<n;i=(i+k)%n){
        e--;
        if(c[i]==1){
            e-=2;
        }
        if(i==0){
            break;
        }
    }
    }
    return e;

}

function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const nk = readLine().split(' ');

    const n = parseInt(nk[0], 10);

    const k = parseInt(nk[1], 10);

    const c = readLine().split(' ').map(cTemp => parseInt(cTemp, 10));

    let result = jumpingOnClouds(c, k);

    ws.write(result + "\n");

    ws.end();
}

File: Jumping_on_the-Clouds_Revisited.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the jumpingOnClouds function below.
def jumpingOnClouds(c, k):
    
    clouds = [c[0]]
    jumps = 1    
    
    while (1 + (jumps * k)) % n != 1:
        clouds.append(c[(jumps * k) % n])        
        jumps += 1
        
    return 100 - jumps - (sum(clouds)*2)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    nk = input().split()

    n = int(nk[0])

    k = int(nk[1])

    c = list(map(int, input().rstrip().split()))

    result = jumpingOnClouds(c, k)

    fptr.write(str(result) + '\n')

    fptr.close()
